sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionLoadBalancerImpl.java

```diff
@@ -111,30 +111,30 @@ private Mono<Void> run(CancellationToken cancellationToken) {
                         if (cancellationToken.isCancellationRequested()) return Mono.empty();
                         return this.partitionController.addOrUpdateLease(lease);
                     })
-                    .then(Mono.just(this)
-                        .flatMap(value -> {
-                            if (cancellationToken.isCancellationRequested()) {
-                                return Mono.empty();
-                            }
-
-                            Instant stopTimer = Instant.now().plus(this.leaseAcquireInterval);
-                            return Mono.just(value)
-                                .delayElement(Duration.ofMillis(100), CosmosSchedulers.COSMOS_PARALLEL)
-                                .repeat( () -> {
-                                    Instant currentTime = Instant.now();
-                                    return !cancellationToken.isCancellationRequested() && currentTime.isBefore(stopTimer);
-                                }).last();
-                        })
-                    );
+                    .then();
             })
             .onErrorResume(throwable -> {
                 // "catch all" exception handler to keep the loop going until the user stops the change feed processor
                 logger.warn("Unexpected exception thrown while trying to acquire available leases", throwable);
                 return Mono.empty();
             })
-            .repeat(() -> {
-                return !cancellationToken.isCancellationRequested();
-            })
+            .then(
```

Comment:
I wonder if placing this above `onErrorResume()` or below makes any difference? Is that what we are doing here?